### PR TITLE
Add min_count and max_count options to ec2 module

### PR DESCRIPTION
**Pull request by wrouesnel**
These two options fulfill the same purpose as the exact_count option, but allow
for loosely specifying the number of instances or bounding it.

min_count ensures at least a certain number of instances are running, max_count
ensures no more then a certain number are running.

min_count and max_count can be used together, but are mutually exclusive to
exact_count (which is functionally identical to setting both values to the same
number).

These options make performing non Amazon autoscaling much easier, and
in particular allow idempotent deploy and bootstrapping ("ensure at least 1
production machine exists" can now be written into a play.)

### CODE REVIEW

**Comment by robynbergeron** [comment]
Hi @wrouesnel -- thanks for submitting this PR.

@tgerla @lwade -- Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

Also pinging the AWS crowd en masse for eyeballs -- @jsmartin @scicoin-project @tombamford @garethr @lorin @jarv @jsdalton @silviud @adq @zbal @zeekin @willthames @carsongee @defionscode @tastychutney @bpennypacker @loia

File: cloud/amazon/ec2.py [hunk 1]
```diff
@@ -233,6 +233,20 @@
     required: false
     default: null
     aliases: []
+  min_count:
+    version_added: "2.0"
+    description:
+      - An integer value which indicates the minimum number of instances that match the 'count_tag' parameter should be running. Instances are created if less then this number are running.
+    required: false
+    default: null
+    aliases: []
```
Comment (juliedavila) [inline cloud/amazon/ec2.py]: Plz remove the `[]` aliases

**Comment by juliedavila** [comment]
This looks good, but this begs the question, why would you want to use this instead of an auto scaling group which carried the first class citizen support from AWS?

**Comment by wrouesnel** [comment]
I created this to deal with scaling up a database cluster which shouldn't really be scaled down. It was easier to setup the playbook so it would bootstrap the cluster if not there (for dev purposes - i.e. min 1) but then if I've added more nodes I want it to also run and provision those nodes separately.

It hugely simplifies giving instructions to someone else on how to use it since you can just say "run this" rather then "run this file once, then you need to run this if you want to shut it down etc. etc."

**Comment by wrouesnel** [comment]
Also re: the `[]` aliases - that seems to be the pattern in most of the rest of the help txt in the file?

**Comment by wimnat** [comment]
As per the module checklist, http://docs.ansible.com/ansible/developing_modules.html, empty aliases are not required

**Comment by wrouesnel** [comment]
Empty aliases removed and rebased to latest devel.

**Comment by gregdek** [comment]
Thanks @wrouesnel for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

File: cloud/amazon/ec2.py [hunk 2]
```diff
@@ -1359,6 +1384,8 @@ def main():
             termination_protection = dict(type='bool', default=False),
             state = dict(default='present', choices=['present', 'absent', 'running', 'stopped']),
             exact_count = dict(type='int', default=None),
+            min_count = dict(type='int', default=None),
```
Comment (juliedavila) [inline cloud/amazon/ec2.py:1387]: the default of `min_count` should be 0 here as that is what the code sets

File: cloud/amazon/ec2.py [hunk 3]
```diff
@@ -233,6 +233,18 @@
     required: false
     default: null
     aliases: []
+  min_count:
+    version_added: "2.0"
+    description:
+      - An integer value which indicates the minimum number of instances that match the 'count_tag' parameter should be running. Instances are created if less then this number are running.
+    required: false
+    default: null
```
Comment (juliedavila) [inline cloud/amazon/ec2.py]: default should be 0

**Comment by juliedavila** [comment]
@wrouesnel just update the default value for `min_count` and update `version_added` to 2.1 and we should be good to go

**Comment by wrouesnel** [comment]
Rebased with documentation fixes.

**Comment by gregdek** [comment]
Moving to shipit per @defionscode feedback above. @wrouesnel thanks for the PR!

**Comment by gregdek** [comment]
Thanks @wrouesnel for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**